keras_core/backend/tensorflow/optimizer.py

```diff
@@ -86,8 +86,6 @@ def apply_grad_to_update_var(var, grad):
             return self.update_step(grad, var, learning_rate)
 
         for grad, var in grads_and_vars:
-            if isinstance(var, backend.Variable):
```

Comment:
I suspect this may have been there for a reason. Why remove it?

Reply:
It was actually breaking `stateless_apply` on the tensorflow backend (which maybe we don't care about). The end error was attempt to run `eager_tensor.assign(x)`.

I could definitely just not test `stateless_apply` on tf, but I couldn't figure out why this was here, removing it fixed the issue, and broke no tests. Any ideas on why those two lines?

Reply:
Alternately, I can just only test `stateless_apply` for jax, on only `apply` for tf/torch, that just seemed more janky.

Reply:
SG